backend/app/controllers/api/v1/sessions_controller.rb

```diff
@@ -0,0 +1,25 @@
+module API
+  module V1
+    class SessionsController < BaseController
+      before_action :authenticate_user!, only: [:destroy]
+
+      def create
+        user = User.find_by_email(params[:email])
+
+        if user&.valid_password?(params[:password])
+          # warden checks devise options like confirmable
+          # if not confirmed then throws an error that is handled by failure_app
```

Comment:
I personally believe that this warden check is done even before it comes here --> it somehow detects that session exists and tries to load user. Whatever, quite mystery for me :D